Migrate my-sites/sharing to webpack css pipeline

blowery opened the pull request:
Also remove a couple classes that were unused across calypso.

#### Changes proposed in this Pull Request

* Migrate my-sites/sharing styles to the new webpack css pipeline
* DOES NOT break up the styles to more closely associate them with each component. That can be a follow up PR, if ever.

#### Testing instructions

* Pull up My Sites -> Sharing. Make sure it looks like it did before this PR.


Part of #27515

blowery committed:
Migrate my-sites/sharing to webpack css pipeline

Also remove a couple classes that were unused across calypso.

blowery commented:
I went after my-site/sharing here because it has a _lot_ of CSS. 4kb zipped.

blowery commented:
It also uses @includes to generate specific style rules for the various services, so the generated CSS is likely even larger.

sgomes reviewed (approved):
Thanks, @blowery, great work!

There appear to be some small changes in specificity, which may or may not be a good thing. For example, in `.card.section-header.is-compact`, the padding is currently being overriden by some CSS coming from what looks like a Gutenberg async load, setting it to `11px` top and bottom.

![image](https://user-images.githubusercontent.com/409615/48618234-a6ee5600-e990-11e8-84ad-825a7b391d7f.png)

With this change, `devdocs.css` takes precedence and changes things:

![image](https://user-images.githubusercontent.com/409615/48618286-cbe2c900-e990-11e8-985b-c4cb97b98ffb.png)

This is not in any way a breaking change, and it's only really visible when doing a side-by-side comparison, but I figured I should point it out in case it was a symptom of something worse.

Other than this, the changes look good, so I'm approving the PR and letting you handle what I pointed out as you see fit :) Do let me know if you want me to take another look at any point, though!

blowery commented:
@sgomes nice catch! neither of those things should be leaking so we should probably fix 'em both... I'll take a look.

blowery committed:
Use a sass comment to prevent this from outputing real CSS comments.
Appears to reduce the number of instances of the section-header css across bundles

jsnajdr commented:
@sgomes I'm unable to reproduce the padding bug you're reporting. (I rebased the branch onto latest master locally, so that might be the reason). There are two CSS rules that affect the heading's vertical padding:
- `@media (min-width: 481px) .card.is-compact` from `components/card` sets the vertical padding to 16px
- `.section-header.card` from  `components/section-header` resets the v-padding to 11px

Both selectors have the same specificity, so order matter. I didn't manage to trigger a scenario where the order would get reversed.

If you're still able to reproduce the bug, can you look at which CSS rules are causing the padding to be 16px and which files are they coming from?

One more thing: it would be nice to migrate also the `my-sites/sharing/connections` styles in this PR. That would do the `sharing` migration in one step and the PR is still very small. @blowery would you mind if I push a commit or two to this branch?

sgomes commented:
@jsnajdr Yes, it seems to be gone now, likely due to the rebase 👍

jsnajdr committed:
Sharing Connections: fix eslint warnings (JSX classnames, React lifecycles)

jsnajdr committed:
Sharing Services Group: remove unused CSS rules

jsnajdr committed:
Migrate my-sites/sharing/connections to webpack CSS

jsnajdr commented:
I pushed a few commits that migrate `my-sites/sharing/connections`, too. This completes the migration of `my-sites/sharing` in one step. There are also a handful of ESLint fixes.

The PR is now ready for testing and landing.

sirreal commented on assets/stylesheets/shared/_extends-forms.scss:
```diff
@@ -1,8 +1,4 @@
-/**
- * SASS placeholder selectors for form styles
- *
- * @format
```
Can we keep the prettier pragma?

jsnajdr replied:
The pragma needs to be in a `/**`-ish comment. But these are then shipped in the production bundle. The change to `//` tries to avoid that. I don't know if there is a way to have both -- pragma that gets removed when bundling for production. Seems there isn't.

jsnajdr commented:
> I went after my-site/sharing here because it has a lot of CSS. 4kb zipped.

The `sharing` webpack CSS chunk that this PR introduces has 27kB minified. That's 2.7% of the 1MB `style.css` (the all-in-one one that we're trying to break up with the webpack migration.)

blowery commented:
LGTM, thanks for the changes @jsnajdr
